src/rpc/mining.cpp

```diff
@@ -1016,17 +1016,6 @@ static RPCHelpMan submitblock()
     }
 
     ChainstateManager& chainman = EnsureAnyChainman(request.context);
-    uint256 hash = block.GetHash();
-    {
-        LOCK(cs_main);
-        const CBlockIndex* pindex = chainman.m_blockman.LookupBlockIndex(hash);
```

Comment:
This is also a debug only log, and it otherwise does not seem to be hitting any hot code paths, so I am not too concerned either.

Reply:
it's unconditional once out of IBD. But allowing untrusted parties access to this rpc is unsafe anyway (can spam the node with low-work blocks since `force_processing` and `min_pow_checked` are true for the `ProcessNewBlock` call).